Mathlib/CategoryTheory/Equivalence.lean

```diff
@@ -766,4 +766,56 @@ noncomputable instance fullyFaithfulToEssImage (F : C ⥤ D) [Full F] [Faithful
 
 end Equivalence
 
+namespace Iso
+
+variable {E : Type u₃} [Category.{v₃} E] {F : C ⥤ E} {G : C ⥤ D} {H : D ⥤ E}
+
+/-- Construct an isomorphism `F ⋙ H.inv ≅ G` from an isomorphism `F ≅ G ⋙ H`. -/
+@[simps!]
+def compInvIso [h : IsEquivalence H] (i : F ≅ G ⋙ H) : F ⋙ H.inv ≅ G :=
+  isoWhiskerRight i H.inv ≪≫
+    associator G H H.inv ≪≫ isoWhiskerLeft G h.unitIso.symm ≪≫ G.rightUnitor
+#align category_theory.comp_inv_iso CategoryTheory.Iso.compInvIso
+
+/-- Construct an isomorphism `F ⋙ H.inverse ≅ G` from an isomorphism `F ≅ G ⋙ H.functor`. -/
+@[simps!]
+def compInverseIso {H : D ≌ E} (i : F ≅ G ⋙ H.functor) : F ⋙ H.inverse ≅ G :=
+  i.compInvIso
```

Comment:
The simp lemmas that are generated for the `Equivalence` versions do not look great. I think it would be better to define `compInverseIso` (and similar definitions) first in the case of equivalences of categories and then define the `IsEquivalence` variants, so that both versions get more appropriate associated simp lemmas.

Reply:
Thanks, I stupidly only checked that the simp lemmas for the `IsEquivalence` versions still looked good after switching away from `eqToIso`... I added the missing rfl-lemma `theorem isEquivalence_unitIso (E : C ≌ D) : IsEquivalence.unitIso = E.unitIso` (and the analogous counit statement) further up in the file and now the simp lemmas are as expected.